imageregistry/v1/types.go

```diff
@@ -303,8 +302,60 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// When type: Internal is used, a vnetName, subNetName and privateEndpointName
+	// may optionally be specified. If unspecificed, the image registry operator
+	// will discover vnet and subnet names, and generate a privateEndpointName.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
```

Comment:
Good practice to introduce unit tests for the new fields and possible cases. E.g. https://github.com/openshift/api/blob/master/machine/v1/stable.controlplanemachineset.aws.testsuite.yaml

Reply:
Added to `imageregistry/v1/stable.config.testsuite.yaml`